appyx-navigation/common/src/iosMain/kotlin/com/bumble/appyx/navigation/platform/LifecycleListener.kt

```diff
@@ -0,0 +1,73 @@
+package com.bumble.appyx.navigation.platform
+
+import com.bumble.appyx.navigation.lifecycle.Lifecycle
+import platform.Foundation.NSNotificationCenter
+import platform.Foundation.NSOperationQueue
+import platform.UIKit.UIApplicationDidBecomeActiveNotification
+import platform.UIKit.UIApplicationDidEnterBackgroundNotification
+import platform.UIKit.UIApplicationWillResignActiveNotification
+import platform.UIKit.UIApplicationWillTerminateNotification
+import platform.darwin.NSObjectProtocol
+
+class LifecycleListener {
+    val lifecycle: PlatformLifecycleRegistry = PlatformLifecycleRegistry()
+
+    private lateinit var didEnterBackgroundNotificationObserver: NSObjectProtocol
+    private lateinit var willResignActiveNotificationObserver: NSObjectProtocol
+    private lateinit var didBecomeActiveNotificationObserver: NSObjectProtocol
+    private lateinit var willTerminateNotificationObserver: NSObjectProtocol
+
+    init {
+        created()
+        startObserving()
+    }
+
+    private fun startObserving() {
+        // GOES TO BACKGROUND
```

Comment:
Please don't use full caps for the comments 😄

Reply:
Done